lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -106,6 +110,42 @@ def workspace_did_change_watched_files(changes)
           constants: constants,
         )
       end
+
+      private
+
+      sig { void }
+      def handle_gemfile_changes
+        return unless File.exist?(".git")
+
+        gemfile_status = %x(git status --porcelain Gemfile.lock).strip
+        return if gemfile_status.empty?
+
+        process_gemfile_changes
+      end
+
+      sig { returns(T.nilable(Integer)) }
+      def process_gemfile_changes
+        current_lockfile = File.read("Gemfile.lock")
+        snapshot_lockfile = File.read(GEMFILE_LOCK_SNAPSHOT) if File.exist?(GEMFILE_LOCK_SNAPSHOT)
+
+        unless snapshot_lockfile
```

Comment:
Have we compared the snapshot approach to `git diff`? I can think of one scenario where you update the lockfile before snapshot is created so upon the consequent restart there are changes in `git status` however, snapshot doesn't exist yet so we don't generate. It's not a common scenario so not important. But feels like running `git diff` instead is a viable solution. Curious about the reasoning here. Does it have downsides for changing branches?